check_test.go

```diff
@@ -157,6 +157,51 @@ func TestIsDialAddr(t *testing.T) {
 	}
 }
 
+func TestURL(t *testing.T) {
+	tests := []struct {
+		in      string
+		wantErr bool
+	}{
+		// Valid
+		{"http://localhost", false},
+		{"http://localhost:1234", false},
+		{"http://192.168.0.1:1234", false},
+		{"http://localhost/home", false},
+		{"http://example.com", false},
+		{"http://example.com/", false},
+		{"http://example.com/home", false},
+		{"https://example.com", false},
+		{"https://example.com/", false},
+		{"https://example.com/home", false},
+		{"https://example.com/home?a=b", false},
+
+		// Invalid
+		{"", true},
+		{" http://foo.com", true},    // invalid character in schema
+		{"ht tp://foo.com", true},    // invalid character in schema
+		{"1http://foo.com", true},    // invalid character in schema
+		{"http://a b.com/", true},    // no space in host name please
+		{"cache_object://foo", true}, // scheme cannot have _, relative path cannot have : in first segment
```

Comment:
nit: I would delete probably all but one or two examples of valid / invalid, as this isn't really testing our code, it's testing the URL parsing code.

Reply:
Done. I kept a few valid combinations just to make it obvious that you can and should be able to use full URLs including paths and query params.